### PR TITLE
P4-1496 Add PATCH endpoint for Profile V2

### PR DESCRIPTION
**Pull request by cwrw**
### Jira link

[P4-1496](https://dsdmoj.atlassian.net/browse/P4-1496)

### What?

I have added/removed/altered:

- [x] Added a new update route for profiles nested under people
- [x] Refactored profile controller slightly

### Why?

To allow users to amend profile assessment answers, add a new endpoint under `/people/{person_id}/profiles/{profile_id}` which takes in assessment answers in the format of jsonb and updates them on the record. I intentionally didn't add any validation for now as this will come in later tickets

I've also updated the missing swagger documentation for POST profiles and added the new endpoint for patch.

### Have you? (optional)

- [x] Updated API docs if necessary

### CODE REVIEW

File: app/controllers/api/v2/profiles_controller.rb [hunk 1]
```diff
@@ -17,7 +24,6 @@ def create
             %i[
               key
               date
-              expiry_data
```
Comment (cwrw) [inline app/controllers/api/v2/profiles_controller.rb:20]: This wasn't being used

Comment (willfish) [reply app/controllers/api/v2/profiles_controller.rb:20]: 👍

File: app/controllers/api/v2/profiles_controller.rb [hunk 2]
```diff
@@ -38,18 +44,26 @@ def profile_params
         params.require(:data).permit(PROFILE_ATTRIBUTES).to_h
       end
 
-      def profile_attributes
+      def new_profile_attributes
```
Comment (willfish) [inline app/controllers/api/v2/profiles_controller.rb:47]: I guess we'll not separate attributes like this later?

File: spec/requests/api/v2/profiles_controller_create_spec.rb [hunk 3]
```diff
@@ -130,6 +130,14 @@
       it_behaves_like 'an endpoint that responds with error 400'
     end
 
+    context 'when the person_id is not found' do
```
Comment (willfish) [inline spec/requests/api/v2/profiles_controller_create_spec.rb:133]: 👍

File: app/controllers/api/v2/profiles_controller.rb (hunk 2 above)
Comment (cwrw) [reply app/controllers/api/v2/profiles_controller.rb:47]: yea as when we revisit this we won't need the first name/last name

File: spec/requests/api/v2/profiles_controller_update_spec.rb [hunk 4]
```diff
@@ -0,0 +1,205 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Api::V2::ProfilesController do
+  let(:response_json) { JSON.parse(response.body) }
```
Comment (willfish) [inline spec/requests/api/v2/profiles_controller_update_spec.rb:6]: I keep wondering whether we want to move to move this into the rails helper or something? It's so common in all of our request specs

File: app/controllers/api/v2/profiles_controller.rb [hunk 5]
```diff
@@ -4,8 +4,15 @@ module Api
   module V2
     class ProfilesController < ApiController
```
Comment (martyn-w) [inline app/controllers/api/v2/profiles_controller.rb:5]: Could you add a TODO to say that validation should be added

File: spec/requests/api/v2/profiles_controller_update_spec.rb [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+  let(:access_token) { create(:access_token).token }
+  let(:content_type) { ApiController::CONTENT_TYPE }
+  let(:headers) { { 'CONTENT_TYPE': content_type }.merge('Authorization' => "Bearer #{access_token}") }
+  let(:risk_type_1) { create :assessment_question, :risk }
```
Comment (willfish) [inline spec/requests/api/v2/profiles_controller_update_spec.rb:10]: Don't usually feel like `<thing>_<n>` is that revealing as far as naming things is concerned. I guess we get value out of creating multiple assessment questions 🤷‍♂️

Comment (willfish) [reply spec/requests/api/v2/profiles_controller_update_spec.rb:10]: Maybe we could create one and call is `assessment_question` and validate against that?

Comment (willfish) [reply spec/requests/api/v2/profiles_controller_update_spec.rb:10]: There's nothing different for one or more assessment questions when updating jsonb?

File: spec/requests/api/v2/profiles_controller_update_spec.rb [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+    context 'when the profile_id is not found' do
+      let(:move_id) { 'foo-bar' }
+      let(:detail_404) { "Couldn't find Profile with 'id'=foo-bar" }
+
+      before { patch "/api/v2/people/#{profile.person.id}/profiles/foo-bar", params: profile_params, headers: headers, as: :json }
+
+      it_behaves_like 'an endpoint that responds with error 404'
+    end
+
+    context 'when the person_id is not found' do
```
Comment (willfish) [inline spec/requests/api/v2/profiles_controller_update_spec.rb:189]: 👍

File: spec/swagger/definitions/hand_coded_paths_v2.yaml [hunk 8]
```diff
@@ -110,3 +110,131 @@
           application/vnd.api+json:
             schema:
               $ref: post_people_responses.yaml#/422
+/people/{person_id}/profiles:
```
Comment (willfish) [inline spec/swagger/definitions/hand_coded_paths_v2.yaml:113]: I guess /api/v2 is shared by a prefix in the overall document :)

File: spec/requests/api/v2/profiles_controller_update_spec.rb (hunk 6 above)
Comment (cwrw) [reply spec/requests/api/v2/profiles_controller_update_spec.rb:10]: not that I'm aware of no, but I think we should add validation for those though, I've discussed this with Jonny, and we'll pick those up in later work next sprint

File: spec/swagger/definitions/hand_coded_paths_v2.yaml (hunk 8 above)
Comment (cwrw) [reply spec/swagger/definitions/hand_coded_paths_v2.yaml:113]: Yea its on the server name itself, and to view this you have to explicitly navigate to the v2 spec:
Screenshot 2020-06-09 at 09 31 21

**Pull request merged**